src/spi_bus.cpp

```diff
@@ -0,0 +1,364 @@
+    size_t position_1 = MotorBoardInterface::MeasurementIndex::position_1;
+    size_t velocity_0 = MotorBoardInterface::MeasurementIndex::velocity_0;
+    size_t velocity_1 = MotorBoardInterface::MeasurementIndex::velocity_1;
+    size_t analog_0 = MotorBoardInterface::MeasurementIndex::analog_0;
+    size_t analog_1 = MotorBoardInterface::MeasurementIndex::analog_1;
+    size_t encoder_index_0 = MotorBoardInterface::MeasurementIndex::encoder_index_0;
+    size_t encoder_index_1 = MotorBoardInterface::MeasurementIndex::encoder_index_1;
+    // receive data from board in a loop ---------------------------------------
+    rt::Spinner spinner;
+    spinner.set_period(0.001); // period of 1ms
+    while(is_loop_active_)
+    {
+        // This will read the last incomming packet and update all sensor fields.
+        main_board_interface_->ParseSensorData();
+    
+        for(size_t i = 0 ; i < nb_udrivers_ ; ++i)
+        {
+            MotorDriver& udriver = main_board_interface_->motor_drivers[i];
+            measurement_[i* measurement_count + current_0]->append(udriver.motor1->GetCurrent());
+            measurement_[i* measurement_count + current_1]->append(udriver.motor2->GetCurrent());
+            measurement_[i* measurement_count + position_0]->append(udriver.motor1->GetPosition() /* * 2 * M_PI */);
+            measurement_[i* measurement_count + position_1]->append(udriver.motor2->GetPosition() /* * 2 * M_PI */);
+            measurement_[i* measurement_count + velocity_0]->append(udriver.motor1->GetVelocity() /* * 2 * M_PI * (1000./60.) */);
+            measurement_[i* measurement_count + velocity_1]->append(udriver.motor2->GetVelocity() /* * 2 * M_PI * (1000./60.) */);
+            measurement_[i* measurement_count + analog_0]->append(udriver.adc[0]);// not implemented yet in the API
+            measurement_[i* measurement_count + analog_1]->append(udriver.adc[1]);// not implemented yet in the API
+            // here the interpretation of the message is different,
+            // we get a motor index and a measurement
+            if(udriver.motor1->GetIndexToggleBit())
+            {
+                measurement_[encoder_index_0]->append(udriver.motor1->GetPosition() /* * 2 * M_PI */);
+            }
+            if(udriver.motor2->GetIndexToggleBit())
+            {
+                measurement_[encoder_index_1]->append(udriver.motor2->GetPosition() /* * 2 * M_PI */);
+            }
+            MotorBoardStatus status;
+            status.system_enabled = udriver.is_enabled;
+            status.motor1_enabled = udriver.motor1->is_enabled;
+            status.motor1_ready   = udriver.motor1->is_ready;
+            status.motor2_enabled = udriver.motor2->is_enabled;
+            status.motor2_ready   = udriver.motor2->is_ready;
+            status.error_code     = udriver.error_code;
+            status_[i]->append(status);
+        }
+    
+        if(!is_ready() && motors_are_paused_)
```

Comment:
not quite sure what's the intention here, but are you sure this should not be an OR instead of an AND?